### PR TITLE
fix: Attempt to fix possible crash when using CGColor

### PR DESCRIPTION
**Pull request by dr1rrb**
closes https://github.com/unoplatform/uno/issues/10283

## Bugfix
Attempt to fix possible crash when using CGColor

## What is the current behavior?
We get some crash reports when changing the `FillColor` of an `CALayer`.

## What is the new behavior?
We are flaggimg the native `CGColor` instances with `GC.KeepAlive` to make sure the ref remains active long enough.

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Shapes/Shape.cs [hunk 1]
```diff
@@ -91,7 +91,7 @@ private void InvalidateForBrushChanged()
 #endif
 			}
 
-#region Stroke Dependency Property
+		#region Stroke Dependency Property
```
Comment (spouliot) [inline src/Uno.UI/UI/Xaml/Shapes/Shape.cs:94]: you might want to exclude the changes to this file

**Review by spouliot — APPROVED** [review]
👍

File: src/Uno.UI/UI/Xaml/Shapes/Shape.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.cs:94]: I might ... but as it was only few lines I thought it was clear enough and it would improve code quality

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/4.4

**Comment by jeromelaban** [comment]
https://github.com/Mergifyio backport release/stable/4.5

**Pull request merged**